spec/api/audio_events_spec.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+require 'swagger_helper'
+
+describe 'audio_events' do
+  create_entire_hierarchy
+
+  sends_json_and_expects_json
+  with_authorization
+  for_model AudioEvent
+  which_has_schema ref(:audio_event)
+  let(:audio_recording_id) { audio_recording.id }
+
+  # PATCH|PUT /audio_recordings/:audio_recording_id/audio_events/:id      audio_events#update {:format=>"json"}
+  # DELETE    /audio_recordings/:audio_recording_id/audio_events/:id      audio_events#destroy {:format=>"json"}
+  # GET       /projects/:project_id/audio_events/download                 audio_events#download {:format=>"csv"}
+  # GET       /projects/:project_id/sites/:site_id/audio_events/download  audio_events#download {:format=>"csv"}
+  # GET       /user_accounts/:user_id/audio_events/download               audio_events#download {:format=>"csv"}
+
+  path '/audio_events/filter' do
+    post('filter audio_event') do
+      response(200, 'successful') do
+        schema_for_many
+        run_test! do
+          expect_at_least_one_item
+        end
+      end
+    end
+  end
+
+  path '/audio_recordings/{audio_recording_id}/audio_events/filter' do
+    with_route_parameter(:audio_recording_id) { audio_recording_id }
+
+    post('filter audio_events') do
+      response(200, 'successful') do
+        schema_for_many
+        run_test! do
+          expect_at_least_one_item
+        end
+      end
+    end
+  end
```

Comment:
i think i'd prefer the shallow and nested routes be split into two separate example groups or even files